corehq/warehouse/models/shared.py

```diff
@@ -0,0 +1,86 @@
+import os
+
+from django.db import connections
+from django.conf import settings
+from django.template import Context, engines
+
+from corehq.sql_db.routers import db_for_read_write
+
+from corehq.warehouse.transforms import get_transform
+from corehq.warehouse.const import (
+    DJANGO_MAX_BATCH_SIZE,
+)
+
+
+class WarehouseTableMixin(object):
+
+    @classmethod
+    def sql_query_template_name(cls):
+        return None
+
+    @classmethod
+    def dependencies(cls):
+        raise NotImplementedError
+
+    @classmethod
+    def load(cls):
+        '''
+        Bulk loads records for a dim or fact table from
+        their corresponding dependencies
+        '''
+        for record in cls.record_iter():
+            print record
+
+    @classmethod
+    def _table_context(cls):
```

Comment:
we need a dictionary to map slugs to table names so we can test queries in django tests